src/Microsoft.Sbom.Api/Workflows/SbomConsolidationWorkflow.cs

```diff
@@ -3,41 +3,76 @@
+    public SbomConsolidationWorkflow(ILogger logger, IConfiguration configuration, IWorkflow<SbomGenerationWorkflow> sbomGenerationWorkflow, ISbomConfigFactory sbomConfigFactory, ISPDXFormatDetector sPDXFormatDetector, IFileSystemUtils fileSystemUtils, IMetadataBuilderFactory metadataBuilderFactory)
     {
         this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
         this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        this.sbomConfigFactory = sbomConfigFactory ?? throw new ArgumentNullException(nameof(sbomConfigFactory));
+        this.sPDXFormatDetector = sPDXFormatDetector ?? throw new ArgumentNullException(nameof(sPDXFormatDetector));
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.metadataBuilderFactory = metadataBuilderFactory ?? throw new ArgumentNullException(nameof(metadataBuilderFactory));
         this.sbomGenerationWorkflow = sbomGenerationWorkflow ?? throw new ArgumentNullException(nameof(sbomGenerationWorkflow));
     }
 
     /// <inheritdoc/>
     public virtual async Task<bool> RunAsync()
     {
-        logger.Information("Placeholder SBOM consolidation workflow executed.");
+        var sbomsToValidate = ArtifactInfoMap.Select(artifact => GetSbomsToValidate(artifact.Key, artifact.Value))
+            .Where(l => l != null)
+            .SelectMany(l => l);
+        if (sbomsToValidate == null || !sbomsToValidate.Any())
+        {
+            logger.Information($"No valid SBOMs detected.");
+            return false;
+        }
 
-        return ValidateSourceSboms() && await GeneratedConsolidatedSbom();
+        return await ValidateSourceSbomsAsync(sbomsToValidate) && await GeneratedConsolidatedSbom();
     }
 
-    private bool ValidateSourceSboms()
+    private IEnumerable<(ISbomConfig config, ArtifactInfo info)> GetSbomsToValidate(string artifactPath, ArtifactInfo info)
     {
-        // TODO : Implement the source SBOMs.`
-        return true;
+        var manifestDirPath = info?.ExternalManifestDir ?? fileSystemUtils.JoinPaths(artifactPath, Constants.ManifestFolder);
+        var isValidSpdxFormat = sPDXFormatDetector.TryGetSbomsWithVersion(manifestDirPath, out var detectedSboms);
+        if (!isValidSpdxFormat)
+        {
+            logger.Information($"No SBOMs located in {manifestDirPath} of a recognized SPDX format.");
+            return null;
+        }
+
+        return detectedSboms.Select((sbom) => (sbomConfigFactory.Get(sbom.manifestInfo, manifestDirPath, metadataBuilderFactory), info));
+    }
+
+    private async Task<bool> ValidateSourceSbomsAsync(IEnumerable<(ISbomConfig config, ArtifactInfo info)> sbomsToValidate)
+    {
+        var validationWorkflows = sbomsToValidate
+            .Select(async sbom => await Task.FromResult(true)); // TODO: Run validation workflow
```

Comment:
I love the idea of running the validations concurrently. Two thoughts (probably not for this PR, though):
1. Would it make sense to move the concurrency on level up, so that we're also running the detectors (and trying to parse the files) in parallel?
2. Logging from parallel activities can be difficult to decipher, and some of the telemetry that we capture may need to be broken out by SPDX format. That might impact the design for concurrency.

Ultimately, I wonder if V1 might be better without concurrency, then we can gather data to determine if the performance win of making it concurrent justifies the extra complexith